File: ch8/exec1.c
#include<stdio.h>
#include<unistd.h>
int main(int argc, char const *argv[])
{
    char *arglist[3];
    arglist[0] = "ls";
    arglist[1] = "-l";
    arglist[2] = 0;
    printf("*** about to exec ls -l\n");
    execvp("ls",arglist);
    printf("*** ls is done.bye\n");
    return 0;
}

File: ch8/ch8_src/src/bin/exec1.rs
use std::{error::Error, ffi::CString};

use nix::unistd::execvp;

fn main() -> Result<(), Box<dyn Error>> {
    let arglist: Vec<CString> = vec!["ls", "-l"]
        .into_iter()
        .map(|s| CString::new(s).unwrap())
        .collect();

    println!("*** about to exec ls -l");

    let filename = CString::new("ls")?;
    execvp(&filename, &arglist)?;

    println!("*** ls is done.bye");
    Ok(())
}
